app/components/new-myreport.js

```diff
@@ -23,26 +23,18 @@ const PrepositionObject = EmberObject.extend({
 
   value: oneWay('model.id'),
 
-  label: computed('model', 'type', function() {
-    const type = this.type;
-    const model = this.model;
-    return new Promise(resolve => {
-      switch (type) {
-      case 'mesh term':
-        resolve(model.get('name'));
-        break;
-      case 'term':
-        model.get('vocabulary').then(vocabulary => {
-          model.get('titleWithParentTitles').then(titleWithParentTitles => {
-            const title = vocabulary.get('title') + ' > ' + titleWithParentTitles;
-            resolve(title);
-          });
-        });
-        break;
-      default:
-        resolve(model.get('title'));
-      }
-    });
+  label: computed('model', 'type', async function() {
+    const { model, type } = this.getProperties('model', 'type');
+
+    if (type === 'mesh term') {
+      return model.name;
+    } else if (type === 'term') {
+      const vocabulary = await model.vocabulary;
```

Comment:
Think this is ED as well so should be `model.get('vocabulary')`

Reply:
I think I checked all updates and they were coming back the same.

```
// From
something.get('something');

// To
something.something
```

Reply:
I'm worried by https://blog.emberjs.com/2018/02/16/ember-3-1-beta-released.html which says 

> Ember Data returns promise proxy objects when you read an async relationship and from other API. Ember proxy objects, including promise proxies, still require that you call get to read values.

And this is one of those. It may only be working because network is fast or the data is preloaded so it's async but not really.

Reply:
Okay. We could stay on the safe side then. 👍

Reply:
Reverted and addressed.

Reply:
@thomas, still seeing `model.vocabulary` here. did you forget to push after reverting?

Reply:
I guess I did forget to push.